database/src/pool.rs

```diff
@@ -178,6 +179,15 @@ pub trait Connection: Send + Sync {
 
     /// Removes all data associated with the given artifact.
     async fn purge_artifact(&self, aid: &ArtifactId);
+
+    /// Add a jobs to the queue
+    async fn enqueue_commit_job(&self, target: Target, jobs: &[CommitJob]);
+
+    /// Dequeue jobs
+    async fn dequeue_commit_job(&self, machine_id: &str, target: Target) -> Option<CommitJob>;
```

Comment:
This should probably also set the commit job to be in progress, so it should be maybe named something like "start_commit_job"? Or something like that.